docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -0,0 +1,535 @@
+                    "Datetime string must look like this: '2010-12-27T23:59:59-07:00'",
+                },
+            )
+
+        except edgedb.errors.ConstraintViolationError:
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=f"Event name '{event.name}' already exists,",
+            )
+
+        return ResponseData(
+            name=created_event.name,
+            address=created_event.address,
+            schedule=created_event.schedule,
+            host=Host(name=created_event.host.name) if created_event.host else None,
+        )
+
+Like the ``POST /users`` API, here, the incoming and outgoing shape of the data is defined by the ``RequestData`` and ``ResponseData``models respectively. The ``post_events`` function asynchronously inserts the data into the database and returns the fields defined in the ``SELECT`` statement. EdgeQL allows us to perform insertion and selection of data fields at the same time. The exception handling logic validates the shape of the incoming data. For example, just as before, this API will complain if you try to create multiple events with the same. Also, the field ``schedule`` accepts data as an `ISO 8601 <https://en.wikipedia.org/wiki/ISO_8601>` timestamp string. Failing to do so will incur an HTTP 400 (bad request) error.
+
+Here's how you'd create an event:
+
+
+.. code-block:: bash
+    httpx -m POST http://localhost:5000/events \
+          --json '{"name":"Resuscitation", "address":"Britain", "schedule":"1889-07-27T23:59:59-07:00", "host_name":"Mina Murray"}'
+
+That'll return:
+
+::
+    HTTP/1.1 200 OK
+    ...
+    {
+      "name": "Resuscitation",
+      "address": "Britain",
+      "schedule": "1889-07-28T06:59:59+00:00",
+      "host": {
+        "name": "Mina Murray"
+      }
+    }
+
+You can also use the ``GET /events`` endpoint to list and filter the event objects. To locate the ``Resuscitation`` event, you'd use the ``filter_name`` parameter with the GET API as follows:
+
+.. code-block:: bash
+    httpx -m GET http://localhost:5000/events -p 'name' 'Resuscitation'
+
+That'll return:
+
+::
+    HTTP/1.1 200 OK
+    ...
+    {
+      "name": "Resuscitation",
+      "address": "Britain",
+      "schedule": "1889-07-28T06:59:59+00:00",
+      "host": {
+        "name": "Mina Murray"
+      }
+    }
+
+Take a look at the ``app/events.py`` file to see how the ``PUT /events`` and ``DELETE /events`` endpoints are constructed.
+
+
+Browse the endpoints using the native OpenAPI doc
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+FastAPI automatically generates OpenAPI schema from the API endpoints and uses those to build the API docs. While the ``uvicorn`` server is running, go to your browser and head over to `http://localhost:5000/docs <http://locahost:5000/docs>`_. You should see an API navigator like this:
+
+.. image:: https://user-images.githubusercontent.com/30027932/163834402-1bb766d0-a2c4-4fdf-8b0b-9af2ff47a969.png
```

Comment:
We can add this image files to our docs

Reply:
Wouldn't that blow up the repo size? Do we want that?

Reply:
Not significantly, but you're right.

Here's a solution I came up with:

* I created a new branch `pages` for this repo and enabled GitHub pages.
* https://github.com/edgedb/edgedb/tree/pages is it.
* We can add our images (and whatever else) there and serve them from GitHub, for example: https://edgedb.github.io/edgedb/images/guides/fastapi/fastapi1.png

Reply:
cc @colinhacks @jaclarke @vpetrovykh @elprans

Reply:
Oh well, I'm an idiot -- a separate branch for GH pages is still in the same repo so I achieved nothing. :/

Let's host aux images on our website?